### DISCUSSION TITLE
Tools always enabled

### DISCUSSION THREAD

**Discussion by JTHesse** [open-webui/open-webui#4006] [General]
# Bug Report

## Description

**Bug Summary:**
Tools are always enabled for all models

**Steps to Reproduce:**
- Add Tool
- Enable Tool in one Model only
- Go to chat of another Model

**Expected Behavior:**
The Tool should only be enabled for the models in which they are selected

**Actual Behavior:**
Tools are visible in chats for all models

## Environment

- **Open WebUI Version:** 0.3.10
- **Ollama (if applicable):** 0.2.5

- **Operating System:** Ubuntu 20.04
- **Browser (if applicable):** Firefox 

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

![tools](https://github.com/user-attachments/assets/405194ad-b9b0-4dc8-823b-4cff76d806d7)
![tools2](https://github.com/user-attachments/assets/e64d06e9-4619-45a4-8196-d53d0525aa0e)

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by hchasens** [comment]
You might want to post this as an Issue rather than a discussion.

**Reply by JTHesse** [reply]
This was converted from an issue.

**Comment by kekejun** [comment]
Would be great if tools are categorized into 'model tool' and 'session tool'.
The model tool is activated in the workspace model setting, is always on. 
The session tool is alternative for the user to decide, just like now.

**Comment by veleinfobip** [comment]
I would like to report a couple of issues and suggest features regarding tool management in OpenWebUI.

1. **Limited Tool Selection**: I have a total of 6 tools, but I can only select among 3 of them in the UI. Can this be fixed?

2. **Temporary Tool Disabling**: I want to be able to temporarily disable tools instead of deleting them. Is there a way to implement this functionality in the settings?

3. **Active, Idle, and Disabled Tools**: 
   - Currently, tools selected inside model setup are active by default. I would like the option to categorize tools into:
     - **Activated Tools**: ON by default
     - **Idle Tools**: OFF but can be activated as needed
     - **Disabled Tools**: Cannot be selected at all

I believe these changes could significantly improve the user experience of managing tools within OpenWebUI.
